src/test/java/net/jonathangiles/tools/teenyhttpd/chat/ChatServerButUsingAnnotations.java

```diff
@@ -0,0 +1,87 @@
+package net.jonathangiles.tools.teenyhttpd.chat;
+
+import com.google.gson.Gson;
+import net.jonathangiles.tools.teenyhttpd.GsonMessageConverter;
+import net.jonathangiles.tools.teenyhttpd.Message;
+import net.jonathangiles.tools.teenyhttpd.TeenyApplication;
+import net.jonathangiles.tools.teenyhttpd.annotations.*;
+import net.jonathangiles.tools.teenyhttpd.model.ServerSentEventHandler;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ChatServerButUsingAnnotations {
+
+
+    public static void main(String[] args) {
+        System.setProperty("server.port", "80");
+        TeenyApplication.start(ChatServerButUsingAnnotations.class);
+    }
+
+    // collection of all connected users
+    private final Set<String> users = Collections.synchronizedSet(new HashSet<>());
+
+    private final Gson gson = new Gson();
+
+    @Configuration
+    public GsonMessageConverter getGsonConverter() {
+        return new GsonMessageConverter();
+    }
+
+    @ServerEvent("/messages")
+    public ServerSentEventHandler chatMessages() {
+        return ServerSentEventHandler.create();
+    }
+
+    @ServerEvent("/users")
+    public ServerSentEventHandler users() {
+        return ServerSentEventHandler.create();
+    }
+
+    @Get
+    public File getWebRoot() {
+        return new File("src/test/java/net/jonathangiles/tools/teenyhttpd/chat");
+    }
+
+    @Post("/message")
+    public void message(@QueryParam("message") String message,
+                        @EventHandler("chatMessages") ServerSentEventHandler chatMessagesEventHandler) {
```

Comment:
Is this based on the method name on line 34 (i.e. `chatMessages()`)? Should we have a `name` parameter on the `@ServerEvent` annotation that would override this, but by default we take the name?

Reply:
Yes, it its based on the method name, now I just added the name parameter


```java
@ServerEvent(value = "/messages", name = "messages")
    public ServerSentEventHandler chatMessages() {
        return ServerSentEventHandler.create();
    }
```